faunadb-java-dsl/src/main/java/com/faunadb/client/types/Codecs.java

```diff
@@ -0,0 +1,141 @@
+
+  static {
+    CODECS.put(String.class, Codec.STRING);
+    CODECS.put(Boolean.class, Codec.BOOLEAN);
+    CODECS.put(boolean.class, Codec.BOOLEAN);
+    CODECS.put(Long.class, Codec.LONG);
+    CODECS.put(long.class, Codec.LONG);
+    CODECS.put(Integer.class, Codec.INTEGER);
+    CODECS.put(int.class, Codec.INTEGER);
+    CODECS.put(Short.class, Codec.SHORT);
+    CODECS.put(short.class, Codec.SHORT);
+    CODECS.put(Byte.class, Codec.BYTE);
+    CODECS.put(byte.class, Codec.BYTE);
+    CODECS.put(Character.class, Codec.CHAR);
+    CODECS.put(char.class, Codec.CHAR);
+    CODECS.put(Float.class, Codec.FLOAT);
+    CODECS.put(float.class, Codec.FLOAT);
+    CODECS.put(Double.class, Codec.DOUBLE);
+    CODECS.put(double.class, Codec.DOUBLE);
+    CODECS.put(LocalDate.class, Codec.DATE);
+    CODECS.put(HighPrecisionTime.class, Codec.HP_TIME);
+    CODECS.put(Instant.class, Codec.TIME);
+  }
+
+  public static Codec<Object> getDecoder(Class<?> type) {
+    Codec<?> codec = CODECS.get(type);
+    if (codec != null)
+      return (Codec<Object>) codec;
+
+    Function<Value, Object> decoder = DECODERS.get(type);
+
+    if (decoder == null) {
+      decoder = createDecoder(type);
+
+      DECODERS.put(type, decoder);
+    }
+
+    return new ObjectDecoder(decoder);
+  }
+
+  public static Codec<Object> getEncoder(Encoder encoder, Class<?> type) {
+    Codec<?> codec = CODECS.get(type);
+    if (codec != null)
+      return (Codec<Object>) codec;
+
+    EncoderEntryCache encoderEntryCache = ENCODERS.get(type);
+
+    if (encoderEntryCache == null) {
+      encoderEntryCache = createEncoder(type);
+
+      ENCODERS.put(type, encoderEntryCache);
+    }
+
+    return new ObjectEncoder(encoder, encoderEntryCache);
+  }
+
+  private static EncoderEntryCache createEncoder(Class<?> type) {
+    final Properties.Property[] readProperties = Properties.getReadProperties(type);
+
+    return new EncoderEntryCache() {
+      @Override
+      public Value encode(Encoder encoder, Object value) {
+        Map<String, Value> fields = newHashMap();
+
+        for (Properties.Property property : readProperties) {
+          try {
+            fields.put(property.getName(), encoder.encodeImpl(property.get(value)));
+          } catch (Exception ex) {
+            throw new FaunaException(format("Could not encode field \"%s\". Reason: %s", property.getName(), ex.getMessage()));
+          }
+        }
+
+        return new ObjectV(fields);
+      }
+    };
+  }
+
+  private static class ObjectDecoder implements Codec<Object> {
+    private final Function<Value, Object> decoder;
+
+    ObjectDecoder(Function<Value, Object> decoder) {
+      this.decoder = decoder;
+    }
+
+    @Override
+    public Result<Object> decode(Value value) {
+      return Result.success(decoder.apply(value));
+    }
+
+    @Override
+    public Result<Value> encode(Object value) {
+      return null;
```

Comment:
is it necessary? the user will never call this, it's only used internally by the encoder/decoder